ironfish-cli/src/utils/ledger.ts

```diff
@@ -0,0 +1,152 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import { Assert, createRootLogger, Logger } from '@ironfish/sdk'
+import { AccountImport } from '@ironfish/sdk/src/wallet/exporter'
+import TransportNodeHid from '@ledgerhq/hw-transport-node-hid'
+import IronfishApp, {
+  IronfishKeys,
+  ResponseAddress,
+  ResponseProofGenKey,
+  ResponseSign,
+  ResponseViewKey,
+} from '@zondax/ledger-ironfish'
+
+export class Ledger {
+  app: IronfishApp | undefined
+  logger: Logger
+  PATH = "m/44'/1338'/0"
+
+  constructor(logger?: Logger) {
+    this.app = undefined
+    this.logger = logger ? logger : createRootLogger()
+  }
+
+  connect = async () => {
+    const transport = await TransportNodeHid.create(3000, 3000)
+
+    const app = new IronfishApp(transport)
+
+    const appInfo = await app.appInfo()
+    this.logger.debug(appInfo.appName ?? 'no app name')
+
+    if (appInfo.appName !== 'Ironfish') {
+      this.logger.debug(appInfo.appName ?? 'no app name')
+      this.logger.debug(appInfo.returnCode.toString())
+      this.logger.debug(appInfo.errorMessage.toString())
+      throw new Error('Please open the Iron Fish app on your ledger device')
+    }
+
+    if (appInfo.appVersion) {
+      this.logger.debug(`Ironfish App Version: ${appInfo.appVersion}`)
+    }
+
+    this.app = app
+
+    return { app, PATH: this.PATH }
+  }
+
+  getPublicAddress = async () => {
+    if (!this.app) {
+      throw new Error('Connect to Ledger first')
+    }
+
+    const response: ResponseAddress = await this.app.retrieveKeys(
+      this.PATH,
+      IronfishKeys.PublicAddress,
+      false,
+    )
+
+    if (!response.publicAddress) {
+      this.logger.debug(`No public address returned.`)
+      this.logger.debug(response.returnCode.toString())
+      throw new Error(response.errorMessage)
+    }
+
+    return response.publicAddress.toString('hex')
+  }
+
```

Comment:
Why isn't this in the sdk and consumed by CLI? This could potentially be useful for other developers

Reply:
Our philosophy with Zondax/ Chainport is that we want to keep all 3rd party stuff in the application level and not in the sdk.